lib/ranked-model/ranker.rb

```diff
@@ -164,12 +173,26 @@ def update_index_from_position
         end
       end
 
+      def possibly_rebalance_and_position(min, max)
+        return false unless (max - min).between?(-1, 1) # No room at the inn...
+        rebalance_ranks
+        position_at position
+        true
+      end
+
       def rank_at_average(min, max)
-        if (max - min).between?(-1, 1) # No room at the inn...
-          rebalance_ranks
-          position_at position
+        rank_at( ( ( max - min ).to_f / 2 ).ceil + min ) unless possibly_rebalance_and_position(min, max)
+      end
+
+      def rank_with_preferred_spread(min, max)
+        return if possibly_rebalance_and_position(min, max)
+
+        modifier = min < max ? 1 : -1
+
+        if (max - min).abs < ranker.preferred_spread * 2
```

Comment:
Would you be able to explain this one out for me? Especially the significance of `* 2` and `0.3` :)

Reply:
That's a very good question 😀. I can't remember exactly the reason for that `*2`. But from reading it now i'd say it's checking that there's enough space both before and after. And if it's not: the magic number `0.3` enters the stage and draws a rabbit out from the hat. That might have to be removed. It's just another way of minimizing recalculations, since we're using it in a place where users tend to drag'n drop items on top, which causing the space to run out eventually. This just delays that event to happen really.

Makes any sense?